controllers/drplacementcontrol.go

```diff
@@ -462,6 +456,206 @@ func (d *DRPCInstance) getCurrentHomeClusterName(toCluster string, drClusters []
+	}
+
+	return met, nil
+}
+
+// checkRegionalFailoverPrerequisites checks for any RegionalDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkRegionalFailoverPrerequisites(failoverCluster string) (bool, error) {
+	d.setProgression(rmn.ProgressionWaitForStorageMaintenenceActivation)
+
+	if required, activationsRequired := d.requiresRegionalFailoverPrequisites(failoverCluster); required {
+		return d.checkClusterMaintenanceActivation(failoverCluster, activationsRequired)
+	}
+
+	return true, nil
+}
+
+// requiresRegionalFailoverPrequisites checks protected PVCs as reported by the last known Primary cluster
+// to determine if this instance requires failover maintenance modes to be active prior to initiating
+// a failover
+func (d *DRPCInstance) requiresRegionalFailoverPrequisites(failoverCluster string) (
+	bool,
+	map[string]rmn.StorageIdentifiers,
+) {
+	activationsRequired := map[string]rmn.StorageIdentifiers{}
+
+	vrg := d.getLastKnownPrimaryVRG(failoverCluster)
+	if vrg == nil {
+		// TODO: Is this an error, should we ensure at least one VRG is found in the edge cases?
+		// Potentially missing VRG and so stop failover? How to recover in that case?
+		return false, activationsRequired
+	}
+
+	for _, protectedPVC := range vrg.Status.ProtectedPVCs {
+		if len(protectedPVC.StorageIdentifiers.VolumeReplicationClassModes) == 0 {
+			continue
+		}
+
+		if !hasModes(protectedPVC.StorageIdentifiers.VolumeReplicationClassModes, rmn.Failover) {
+			continue
+		}
+
+		// TODO: Assumption is that if there is a mMode then the ReplicationID is a must, err otherwise?
+		key := protectedPVC.StorageIdentifiers.CSIProvisioner + protectedPVC.StorageIdentifiers.ReplicationID
+		if _, ok := activationsRequired[key]; !ok {
+			activationsRequired[key] = protectedPVC.StorageIdentifiers
+		}
+	}
+
+	return len(activationsRequired) != 0, activationsRequired
+}
+
+// getLastKnownPrimaryVRG gets the last known Primary VRG from the cluster that is not the current failoverCluster
+// This is done inspecting VRGs from the MCV reports, and in case not found, fetching it from the s3 store
+func (d *DRPCInstance) getLastKnownPrimaryVRG(failoverCluster string) *rmn.VolumeReplicationGroup {
+	var vrgToInspect *rmn.VolumeReplicationGroup
+
+	for drcluster, vrg := range d.vrgs {
+		if drcluster == failoverCluster {
+			continue
+		}
+
+		if d.isVRGPrimary(vrg) {
+			// TODO: Potentially when there are more than on primary VRGs find the best one?
+			vrgToInspect = vrg
+
+			break
+		}
+	}
+
+	if vrgToInspect == nil {
+		// TODO: Get it from the s3 store
+		return nil
+	}
+
+	return vrgToInspect
+}
+
+// hasModes is a helper routine that checks if a list of modes has the passed in mode
+func hasModes(modes []rmn.MMode, mode rmn.MMode) bool {
```

Comment:
hasMode (singular) instead. We are only looking for one mode inside the list.

Reply:
Done